File: src/main/java/com/qwwuyu/server/ctrl/TestCtrl.java
package com.qwwuyu.server.ctrl;

import com.alibaba.fastjson.JSON;
import com.qwwuyu.server.bean.ResponseBean;
import com.qwwuyu.server.utils.CommUtil;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.Map;

@Controller
@RequestMapping("/test")
public class TestCtrl {
    @RequestMapping(value = "/get", method = RequestMethod.GET)
    public void get(HttpServletRequest request, HttpServletResponse response) {
        post(request, response);
    }

    @RequestMapping(value = "/post", method = RequestMethod.POST)
    public void post(HttpServletRequest request, HttpServletResponse response) {
        Map<String, String[]> map = request.getParameterMap();
        response.setContentType("application/json;charset=utf-8");
        response.setHeader("Pragma", "No-cache");
        response.setHeader("Cache-Control", "no-cache");
        response.setDateHeader("Expires", 0);
        try {
            response.getWriter().write(JSON.toJSONString(new ResponseBean(1, "", map)));
        } catch (IOException ignored) {
        }
    }

    @RequestMapping(value = "/timeout", method = {RequestMethod.POST, RequestMethod.GET})
    public void timeout(HttpServletRequest request, HttpServletResponse response) {
        CommUtil.sleep(100000);
        post(request, response);
    }

    @RequestMapping(value = "/error", method = {RequestMethod.POST, RequestMethod.GET})
    public void error(HttpServletRequest request, HttpServletResponse response) {
        throw new RuntimeException("err");
    }
}

File: app/src/main/kotlin/com/qwwuyu/gs/controller/seldom/TestCtrl.kt
package com.qwwuyu.gs.controller.seldom

import com.google.gson.Gson
import com.qwwuyu.gs.entity.ResponseBean
import com.qwwuyu.lib.utils.CommUtil
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import java.io.IOException
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Controller
@RequestMapping("/test")
class TestCtrl {
    @RequestMapping(value = ["/get"], method = [RequestMethod.GET])
    operator fun get(request: HttpServletRequest, response: HttpServletResponse) {
        post(request, response)
    }

    @RequestMapping(value = ["/post"], method = [RequestMethod.POST])
    fun post(request: HttpServletRequest, response: HttpServletResponse) {
        val map = request.parameterMap
        response.contentType = "application/json;charset=utf-8"
        response.setHeader("Pragma", "No-cache")
        response.setHeader("Cache-Control", "no-cache")
        response.setDateHeader("Expires", 0)
        try {
            response.writer.write(Gson().toJson(ResponseBean(1, "", map)))
        } catch (ignored: IOException) {
        }
    }

    @RequestMapping(value = ["/timeout"], method = [RequestMethod.POST, RequestMethod.GET])
    fun timeout(request: HttpServletRequest, response: HttpServletResponse) {
        CommUtil.sleep(100000)
        post(request, response)
    }

    @RequestMapping(value = ["/error"], method = [RequestMethod.POST, RequestMethod.GET])
    fun error(request: HttpServletRequest?, response: HttpServletResponse?) {
        throw RuntimeException("err")
    }
}